src/libraries/Microsoft.PowerFx.Core/Public/CheckResult.cs

```diff
@@ -25,7 +25,13 @@ public class CheckResult
         /// null if unavailable.  
         /// This is only valid if there are no errors. 
         /// </summary>
-        public HashSet<string> TopLevelIdentifiers { get; set; }
+        public HashSet<string> TopLevelIdentifiers { get; set; }
+
+        /// <summary>
+        /// Spans of all fields that this formula uses. 
+        /// null if unavailable.  
+        /// </summary>
+        public HashSet<Localization.Span> ReferencedIdentifiers { get; set; }
```

Comment:
>Localization.Span [](http://example.com/codeflow?start=23&length=17)

Can we use strings here? 
Strings are simpler to understand and we've already allocated the string object in the symbol table, so for perf concerns, just be sure to return the existing already-allocated object,